src/Account/Service/UserFollowService.ts

```diff
@@ -0,0 +1,334 @@
+import { HttpStatus, Inject, Injectable } from '@nestjs/common';
+import { RedisService } from 'nestjs-redis';
+import { InjectModel } from '@nestjs/mongoose';
+import { User } from '../Schema/UserSchema';
+import { Model } from 'mongoose';
+import BusinessError from '../../Common/BusinessError';
+import UserGeoService from './UserGeoService';
+import { InjectPinoLogger, PinoLogger } from 'nestjs-pino';
+import { ConfigService } from '@nestjs/config';
+import { UserFollow } from '../Schema/UserFollowSchema';
+import { ErrorCode } from '../../ErrorCode';
+import RedisKeys from '../../Common/RedisKeys';
+
+@Injectable()
+export default class UserFollowService {
+  @InjectPinoLogger(UserFollowService.name)
+  private readonly logger: PinoLogger;
+
+  @Inject()
+  private redisService: RedisService;
+
+  @Inject()
+  private userGeoService: UserGeoService;
+
+  @InjectModel(User.name)
+  private userModel: Model<User>;
+
+  @InjectModel(UserFollow.name)
+  private userFollowModel: Model<UserFollow>;
+
+  @Inject()
+  private configService: ConfigService;
+
+  async follow(userId: string, targetId: string) {
+    // find in cache
+    const queryDbUser = [];
+    if (
+      await this.redisService.getClient().exists(RedisKeys.UserCache(userId))
+    ) {
+      queryDbUser.push(userId);
+    }
+
+    if (
+      await this.redisService.getClient().exists(RedisKeys.UserCache(targetId))
+    ) {
+      queryDbUser.push(targetId);
+    }
+
+    if (queryDbUser.length > 0) {
```

Comment:
If there is no cache data in redis, then it will query the database

Reply:
> then it will query the database

can you point me where is the query code line?

Reply:
> > then it will query the database
> 
> can you point me where is the query code line?

[src/Account/Service/UserFollowService.ts L51](https://github.com/Wiredcraft/test-backend/blob/6a3fa1507894e4723e9b641c458fb62f5d48825d/src/Account/Service/UserFollowService.ts#L51)

Reply:
hmm, but if the `queryDbUser` is empty, it won't query the DB, right? it could be empty if the cached data expires.
https://github.com/Wiredcraft/test-backend/blob/6a3fa1507894e4723e9b641c458fb62f5d48825d/src/Account/Service/UserFollowService.ts#L49

Reply:
This is a more complex consistency problem. Regarding the replacement of the cache, the method of eliminating the cache and then updating the data can be used to ensure data consistency, but it will sacrifice some read performance. If it is simple to think about, you can directly query the database.